pandas/core/reshape.py

```diff
@@ -1280,7 +1320,10 @@ def get_empty_Frame(data, sparse):
     else:
         dummy_mat = np.eye(number_of_cols, dtype=np.uint8).take(codes, axis=0)
 
-        if not dummy_na:
+        if dummy_na == 'keep':
+            dummy_mat = dummy_mat.astype(np.float16)
```

Comment:
If I have a DataFrame like

```python
df = pd.DataFrame({"A": ['a', 'b', np.nan], 'B': [1, 2, 3]})
```

What is the result `.dtypes`? Does it force all columns to float16, even if there' wasn't a NaN in that set (e.g. 'B').

Also, this will force it to float16, even if there were no NaNs to begin with.